### PR TITLE
Add validation for Tool annotation names

### LINKED ISSUE #3832: Tool annotation name validation
It seems there isn't validation around tool names. Certain invalid names can cause LLMs to fail.

**Expected Behavior**
Some type of `@Tool` annotation value validation is done to preemptively catch problems

**Current Behavior**
An invalid tool name (such as having a space in the name) is permitted

**Context**

An invalid tool name (such as having a space in the name) causes the LLM to fail (GPT 4.1-mini in this case).

> Expected a string that matches the pattern '^[a-zA-Z0-9_\\.-]+$'."

### PR DESCRIPTION
**Pull request by academey**
## Summary

This PR adds validation for tool names in the `@Tool` annotation to prevent runtime failures when Large Language Models (LLMs) encounter invalid characters.

## Problem

As reported in #3832, tool names containing spaces or special characters can cause LLMs like GPT 4.1-mini to fail during execution. Currently, there is no validation to prevent these invalid names from being used.

## Solution

- Added regex pattern validation in `ToolUtils.getToolName()` method
- Tool names can now only contain:
  - Alphanumeric characters (a-z, A-Z, 0-9)
  - Underscores (_)
  - Hyphens (-)
  - Dots (.)
- Throws `IllegalArgumentException` with a clear error message when invalid characters are detected
- Updated `@Tool` annotation documentation to clarify naming constraints

## Changes

1. **ToolUtils.java**: Added validation logic with regex pattern `^[a-zA-Z0-9_\\.-]+$`
2. **Tool.java**: Enhanced JavaDoc with naming constraints and examples
3. **ToolUtilsTests.java**: Added comprehensive unit tests covering:
   - Valid tool names
   - Invalid names with spaces
   - Invalid names with special characters
   - Invalid names with parentheses
   - Edge cases (empty names, methods without annotations)

## Testing

- All new tests pass successfully
- Existing tests continue to work
- No breaking changes for valid tool names

## Example

```java
// Valid tool names
@Tool(name = "get_weather")
@Tool(name = "search-docs")
@Tool(name = "tool.v1")

// Invalid tool names (will throw IllegalArgumentException)
@Tool(name = "get weather")  // contains space
@Tool(name = "tool()")       // contains parentheses
@Tool(name = "tool@name!")   // contains special characters
```

Fixes #3832

Signed-off-by: Hyunjoon Park <academey@gmail.com>

### CODE REVIEW

File: spring-ai-model/src/main/java/org/springframework/ai/tool/support/ToolUtils.java [hunk 1]
```diff
@@ -38,16 +39,27 @@
  */
 public final class ToolUtils {
 
+	/**
+	 * Regular expression pattern for valid tool names. Tool names can only contain
+	 * alphanumeric characters, underscores, hyphens, and dots.
+	 */
+	private static final Pattern TOOL_NAME_PATTERN = Pattern.compile("^[a-zA-Z0-9_\\.-]+$");
```
Comment (sunyuhan1998) [inline spring-ai-model/src/main/java/org/springframework/ai/tool/support/ToolUtils.java]: Hi, thank you for paying attention to this issue and trying to resolve it. I admit that such a restriction is a standardized practice, but considering that `Tool` is a generic annotation — it doesn't serve solely openAI or a specific model — limiting tool names to [a-z][A-Z] might be unfair to models that support a wide variety of tool names. This is just my personal opinion; I hope it doesn't bother you. Thank you again.

Comment (academey) [reply spring-ai-model/src/main/java/org/springframework/ai/tool/support/ToolUtils.java]: Thank you for the valuable feedback! 
I agree that enforcing a strict validation pattern could be too restrictive for models that support broader character sets. Let me propose a more flexible approach:

Warning-based approach
 ```java
  // Log a warning instead of throwing an exception
  if (!RECOMMENDED_PATTERN.matcher(toolName).matches()) {
      logger.warn("Tool name '{}' may not be compatible with some LLMs (e.g., OpenAI). " +
                  "Consider using only alphanumeric characters, underscores, hyphens, and dots.",
                  toolName);
  }
```
What do you think about these approach? I'm happy to update the PR

Comment (sunyuhan1998) [reply spring-ai-model/src/main/java/org/springframework/ai/tool/support/ToolUtils.java]: > What do you think about these approach? I'm happy to update the PR

I think the approach you mentioned is a good idea, at least for now, until we have a more refined process and conduct dedicated tool name compatibility checks for each model. It can at least provide users with some troubleshooting ideas in case they really encounter issues with tool name incompatibility.

Comment (academey) [reply spring-ai-model/src/main/java/org/springframework/ai/tool/support/ToolUtils.java]: I've updated the implementation based on the feedback. The validation now logs a warning instead of throwing an exception, making it more flexible while still providing guidance to users.

This approach maintains backward compatibility and allows the Tool annotation to remain generic and LLM-agnostic, while still helping users avoid potential compatibility issues.

Thanks for your feedback @sunyuhan1998
